lib/Db/CardMapper.php

```diff
@@ -547,11 +557,17 @@ public function isOwner($userId, $cardId): bool {
 	}
 
 	public function findBoardId($id): ?int {
-		$sql = 'SELECT id FROM `*PREFIX*deck_boards` WHERE `id` IN (SELECT board_id FROM `*PREFIX*deck_stacks` WHERE id IN (SELECT stack_id FROM `*PREFIX*deck_cards` WHERE id = ?))';
-		$stmt = $this->db->prepare($sql);
-		$stmt->bindParam(1, $id, \PDO::PARAM_INT);
-		$stmt->execute();
-		return $stmt->fetchColumn() ?? null;
+		//$this->cache->remove('findBoardId:' . $id);
+		$result = $this->cache->get('findBoardId:' . $id);
+		if ($result === null) {
+			$sql = 'SELECT id FROM `*PREFIX*deck_boards` WHERE `id` IN (SELECT board_id FROM `*PREFIX*deck_stacks` WHERE id IN (SELECT stack_id FROM `*PREFIX*deck_cards` WHERE id = ?))';
```

Comment:
btw you can put this in query builders as some point:
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/private/Repair/RemoveLinkShares.php#L115

Reply:
@nickvergessen Will do here right away then. Thinking about the query, weren't joins usually also a bit faster than sub-queries?

Reply:
Yeah, looking at this query if there is always a 1:n (card to stack and stack to board) relation, you could optimise for reading and add a board id column to the cards table as well and then save 2 queries?

Reply:
Moved to a single join query for now as introducing a dedicated board_id column would be a larger topic with possibly heavy migrations involved depending on the table size, should already be a good improvement now with the additional caching in place.